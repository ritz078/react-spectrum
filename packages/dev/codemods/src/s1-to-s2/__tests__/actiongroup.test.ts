// @ts-ignore
import {defineSnapshotTest} from 'jscodeshift/dist/testUtils';
import transform from '../src/codemods/codemod';

const test = (name: string, input: string) => {
  defineSnapshotTest(transform, {}, input, name);
};

test('Converts ActionGroup to ActionButtonGroup', `
import {ActionGroup, Item} from '@adobe/react-spectrum';
<ActionGroup onAction={onAction}>
  <Item key="add">Add</Item>
  <Item key="delete">Delete</Item>
  <Item key="edit">Edit</Item>
</ActionGroup>
`);

test('Converts ActionGroup to ActionButtonGroup with disabledKeys', `
import {ActionGroup, Item} from '@adobe/react-spectrum';

function Example({disabledKeys}) {
  return (
    <ActionGroup onAction={onAction} disabledKeys={disabledKeys}>
      <Item key="add">Add</Item>
      <Item key="delete">Delete</Item>
      <Item key="edit">Edit</Item>
    </ActionGroup>
  );
}
`);

test('Converts ActionGroup to ActionButtonGroup with dynamic collections', `
import {ActionGroup, Item} from '@adobe/react-spectrum';
<ActionGroup onAction={onAction} items={items}>
  {item => <Item>{item.name}</Item>}
</ActionGroup>
`);

test('Converts ActionGroup to ActionButtonGroup with dynamic collections and custom key', `
import {ActionGroup, Item} from '@adobe/react-spectrum';
<ActionGroup onAction={onAction} items={items}>
  {item => <Item key={item._id}>{item.name}</Item>}
</ActionGroup>
`);

test('Converts ActionGroup to ToggleButtonGroup', `
  import {ActionGroup, Item} from '@adobe/react-spectrum';
<ActionGroup selectionMode="single" onSelectionChange={onSelectionChange}>
  <Item key="add">Add</Item>
  <Item key="delete">Delete</Item>
  <Item key="edit">Edit</Item>
</ActionGroup>
`);

test('Converts ActionGroup to ToggleButtonGroup with disabledKeys', `
import {ActionGroup, Item} from '@adobe/react-spectrum';

function Example({disabledKeys}) {
  return (
    <ActionGroup selectionMode="single" onSelectionChange={onSelectionChange} disabledKeys={disabledKeys}>
      <Item key="add">Add</Item>
      <Item key="delete">Delete</Item>
      <Item key="edit">Edit</Item>
    </ActionGroup>
  );
}
`);

test('Converts ActionGroup to ToggleButtonGroup with dynamic collections', `
  import {ActionGroup, Item} from '@adobe/react-spectrum';
<ActionGroup selectionMode="single" onSelectionChange={onSelectionChange} items={items}>
  {item => <Item>{item.name}</Item>}
</ActionGroup>
`);

test('Converts ActionGroup to ToggleButtonGroup with dynamic collections and custom key', `
  import {ActionGroup, Item} from '@adobe/react-spectrum';
<ActionGroup selectionMode="single" onSelectionChange={onSelectionChange} items={items}>
  {item => <Item key={item._id}>{item.name}</Item>}
</ActionGroup>
`);

test('Comments out unsupported props', `
import {ActionGroup, Item} from '@adobe/react-spectrum';
<ActionGroup overflowMode="collapse" buttonLabelBehavior="collapse" summaryIcon={<Icon />}>
  <Item key="add">Add</Item>
  <Item key="delete">Delete</Item>
  <Item key="edit">Edit</Item>
</ActionGroup>
`);
