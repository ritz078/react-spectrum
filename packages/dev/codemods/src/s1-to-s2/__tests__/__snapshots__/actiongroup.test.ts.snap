// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments out unsupported props 1`] = `
"import { ActionButtonGroup, ActionButton } from "@react-spectrum/s2";
// TODO(S2-upgrade): overflowMode has not been implemented yet.
// TODO(S2-upgrade): buttonLabelBehavior has not been implemented yet.
// TODO(S2-upgrade): summaryIcon has not been implemented yet.
<ActionButtonGroup>
  <ActionButton key="add">Add</ActionButton>
  <ActionButton key="delete">Delete</ActionButton>
  <ActionButton key="edit">Edit</ActionButton>
</ActionButtonGroup>"
`;

exports[`Converts ActionGroup to ActionButtonGroup 1`] = `
"import { ActionButtonGroup, ActionButton } from "@react-spectrum/s2";
<ActionButtonGroup>
  <ActionButton key="add" onPress={() => onAction("add")}>Add</ActionButton>
  <ActionButton key="delete" onPress={() => onAction("delete")}>Delete</ActionButton>
  <ActionButton key="edit" onPress={() => onAction("edit")}>Edit</ActionButton>
</ActionButtonGroup>"
`;

exports[`Converts ActionGroup to ActionButtonGroup with disabledKeys 1`] = `
"import { ActionButtonGroup, ActionButton } from "@react-spectrum/s2";

function Example({disabledKeys}) {
  let _disabledKeys = new Set(disabledKeys);
  return (
    (<ActionButtonGroup>
      <ActionButton
        key="add"
        onPress={() => onAction("add")}
        isDisabled={_disabledKeys.has("add")}>Add</ActionButton>
      <ActionButton
        key="delete"
        onPress={() => onAction("delete")}
        isDisabled={_disabledKeys.has("delete")}>Delete</ActionButton>
      <ActionButton
        key="edit"
        onPress={() => onAction("edit")}
        isDisabled={_disabledKeys.has("edit")}>Edit</ActionButton>
    </ActionButtonGroup>)
  );
}"
`;

exports[`Converts ActionGroup to ActionButtonGroup with dynamic collections 1`] = `
"import { ActionButtonGroup, ActionButton } from "@react-spectrum/s2";
<ActionButtonGroup>
  {items.map(
    item => <ActionButton key={item.key ?? item.id} onPress={() => onAction(item.key ?? item.id)}>{item.name}</ActionButton>
  )}
</ActionButtonGroup>"
`;

exports[`Converts ActionGroup to ActionButtonGroup with dynamic collections and custom key 1`] = `
"import { ActionButtonGroup, ActionButton } from "@react-spectrum/s2";
<ActionButtonGroup>
  {items.map(
    item => <ActionButton key={item._id} onPress={() => onAction(item._id)}>{item.name}</ActionButton>
  )}
</ActionButtonGroup>"
`;

exports[`Converts ActionGroup to ToggleButtonGroup 1`] = `
"import { ToggleButtonGroup, ToggleButton } from "@react-spectrum/s2";
<ToggleButtonGroup selectionMode="single" onSelectionChange={onSelectionChange}>
  <ToggleButton id="add">Add</ToggleButton>
  <ToggleButton id="delete">Delete</ToggleButton>
  <ToggleButton id="edit">Edit</ToggleButton>
</ToggleButtonGroup>"
`;

exports[`Converts ActionGroup to ToggleButtonGroup with disabledKeys 1`] = `
"import { ToggleButtonGroup, ToggleButton } from "@react-spectrum/s2";

function Example({disabledKeys}) {
  let _disabledKeys = new Set(disabledKeys);
  return (
    (<ToggleButtonGroup selectionMode="single" onSelectionChange={onSelectionChange}>
      <ToggleButton id="add" isDisabled={_disabledKeys.has("add")}>Add</ToggleButton>
      <ToggleButton id="delete" isDisabled={_disabledKeys.has("delete")}>Delete</ToggleButton>
      <ToggleButton id="edit" isDisabled={_disabledKeys.has("edit")}>Edit</ToggleButton>
    </ToggleButtonGroup>)
  );
}"
`;

exports[`Converts ActionGroup to ToggleButtonGroup with dynamic collections 1`] = `
"import { ToggleButtonGroup, ToggleButton } from "@react-spectrum/s2";
<ToggleButtonGroup selectionMode="single" onSelectionChange={onSelectionChange}>
  {items.map(
    item => <ToggleButton key={item.key ?? item.id} id={item.key ?? item.id}>{item.name}</ToggleButton>
  )}
</ToggleButtonGroup>"
`;

exports[`Converts ActionGroup to ToggleButtonGroup with dynamic collections and custom key 1`] = `
"import { ToggleButtonGroup, ToggleButton } from "@react-spectrum/s2";
<ToggleButtonGroup selectionMode="single" onSelectionChange={onSelectionChange}>
  {items.map(
    item => <ToggleButton key={item._id} id={item._id}>{item.name}</ToggleButton>
  )}
</ToggleButtonGroup>"
`;
